Add information on using Hopper on how to workout SwiftUI signatures and creating our glue functions.

CartBlanche opened the pull request.

CartBlanche committed:
Add Details on using Hopper when creating our glue functions.

CartBlanche committed:
Added extra section detailing how to workout native signatures.

CartBlanche committed:
Formatting fixes.

CartBlanche committed:
More Formatting fixes.

CartBlanche committed:
Indented code correctly?

CartBlanche committed:
Trying to get indentation right.

chkn commented on Hacking.md:
```diff
@@ -24,3 +24,128 @@ When P/Invoking a Swift function, you must keep in mind that Swift may pass some
+    * where rdx represents the `*type metadata for SwiftUI.Text`
+    * where rcs represents the `*type metadata for SwiftUI.Color`
+Also worth noting that...
+    * 8 holds the `*protocol witness table for SwiftUI.Text`
```
typo?
```suggestion
    * r8 holds the `*protocol witness table for SwiftUI.Text`
```

CartBlanche replied:
Yes. Fixed.

chkn commented on Hacking.md:
```diff
@@ -24,3 +24,128 @@ When P/Invoking a Swift function, you must keep in mind that Swift may pass some
+Also worth noting that...
+    * 8 holds the `*protocol witness table for SwiftUI.Text`
+    * r9 holds the `*protocol witness table for SwiftUI.Color`
+These are hidden parameters which we'll need later.
```
Should we also perhaps link to the register usage doc in the Swift repo here?

CartBlanche replied:
Yes good call!

CartBlanche replied:
I've also added a link to the SwiftUI.tbd mangled API list.

chkn reviewed (approved):
This is great, thanks!

CartBlanche committed:
Added links t the Architecture Register list as well as full mangled named lists. Also fixed a typo.

chkn committed:
Minor markdown reformatting for Hacking guide

- Prefer explicit code fences with syntax highlighting
- Remove extra indentation
- Minor typos and formatting
